go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/fluxcd/flux2
 
-go 1.17
+go 1.18
```

Comment:
I would not force this to 1.18, but instead keep compatibility at 1.17 for some time. This also guards against people trying to introduce generics in contributions.